postgres-appliance/scripts/configure_spilo.py

```diff
@@ -106,7 +107,7 @@ def write_certificates(environment, overwrite):
         output, _ = p.communicate()
         logging.debug(output)
 
-    uid = os.stat(environment['PGHOME']).st_uid
+    uid = pwd.getpwnam(environment['PGUSER_SUPERUSER']).pw_uid
```

Comment:
This one seems to be totally unnecessary, the original code is producing the same result.
```
$ docker run -u 1001 --rm -ti --entrypoint /bin/bash registry.opensource.zalan.do/acid/spilo-12:1.6-p2 
I have no name!@48fb2b152229:/home/postgres$ python3
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.stat('/home/postgres').st_uid
101
>>>
```

Reply:
The reason behind this was to obtain the UID from the system not the filesystem.
Once `/etc/passwd` is set, results are different:

```
$ docker run -ti --rm --user 1000320000 --entrypoint /bin/bash toronz/spilo-min-cdp-12:1.6-v1.3.9.9
I have no name!@2ababdef33fa:/home/postgres$ python3
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.stat('/home/postgres').st_uid
101
>>> import pwd
>>> pwd.getpwnam('postgres').pw_uid
101
>>> exit()
I have no name!@2ababdef33fa:/home/postgres$ if [ "$(id -u)" -ne 0 ]; then
>     sed -e "s/^postgres:x:[^:]*:[^:]*:/postgres:x:$(id -u):$(id -g):/" /etc/passwd > /tmp/passwd
>     cat /tmp/passwd > /etc/passwd
>     rm /tmp/passwd
>     mkdir -p "$PGROOT"
> fi
I have no name!@2ababdef33fa:/home/postgres$ python3
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.stat('/home/postgres').st_uid
101
>>> import pwd
>>> pwd.getpwnam('postgres').pw_uid
1000320000
```

I used my image based on this PR, due to the permissions on /etc/passwd.